lib/widget/carousel.dart

```diff
@@ -142,6 +145,9 @@ class CarouselState extends WidgetState<Carousel> with TemplatedWidgetState {
   @override
   Widget buildWidget(BuildContext context) {
     ScopeManager? scopeManager = DataScopeWidget.getScope(context);
+
+    customIndicator = defineIndicatorWidget(scopeManager!);
```

Comment:
You have to understand why we don't want to do `!` - don't just move code around. Please spend a few minutes to understand why.